src/System.Text.JsonLab/System/Text/Json/JsonWriter.cs

```diff
@@ -77,6 +78,7 @@ private void WriteStartUtf8(int bytesNeeded, byte token)
 
             byteBuffer[idx] = token;
             _bufferWriter.Advance(bytesNeeded);
+            _bufferWriter.Flush();
```

Comment:
I don't think we can flush on every write. Flush is very expensive when called on Pipelines IBufferWriter. 

cc: @pakrym, @davidfowl

Reply:
Agreed